File: 2015/Day_05/Ruby/test_part_2.rb
require_relative "part_2"
require 'test/unit'

class TestNiceNaughty < Test::Unit::TestCase
    
    def test_rule_one
        assert_equal(true, rule_one('xyxy'))
        assert_equal(true, rule_one('aabcdefgaa'))
        assert_equal(false, rule_one('aaa'))
    end
    
    def test_rule_two
        assert_equal(true, rule_two('xyx'))
        assert_equal(true, rule_two('abcdefeghi'))
        assert_equal(true, rule_two('aaa'))
    end
    
    def test_naughty_or_nice
        assert_equal(true, naughty_or_nice('qjhvhtzxzqqjkmpb'))
        assert_equal(true, naughty_or_nice('xxyxx'))
        assert_equal(false, naughty_or_nice('uurcxstgmygtbstg'))
        assert_equal(false, naughty_or_nice('ieodomkazucvgmuy'))
    end
end

File: 2015/Day_05/Python/test_part_2.py
from . import part_2


def test_rule_one():
    assert part_2.rule_one('xyxy') is True
    assert part_2.rule_one('aabcdefgaa') is True
    assert part_2.rule_one('aaa') is False


# def test_rule_two():
#     assert part_2.rule_two('xx') is True
#     assert part_2rule_two("abcdde (dd)") is True
#     assert part_2.rule_two("aabbccdd") is True
#     assert part_2.rule_two('a') is False
#
#
# def test_naughty_or_nice():
#     assert part_2.naughty_or_nice('ugknbfddgicrmopn') is True
#     assert part_2.naughty_or_nice('aaa') is True
#     assert part_2.naughty_or_nice('jchzalrnumimnmhp') is False
#     assert part_2.naughty_or_nice('haegwjzuvuyypxyu') is False
#     assert part_2.naughty_or_nice('dvszwmarrgswjxmb') is False
